app/models/job.rb

```diff
@@ -12,39 +15,81 @@ class Job < ApplicationRecord
 
   validates :status, presence: true, inclusion: { in: STATUSES }
   validates :kind, presence: true, inclusion: { in: KINDS }
+  validates :end_time, comparison: { greater_than: :start_time }, if: ->{ end_time.present? }
 
   STATUSES.each do |status_name|
+    # A helper method to determine the status of a job
```

Comment:
is this the same thing as defining the status as enum?

Reply:
Yeah it is! I'll switch it over to be an enum